### ISSUE TITLE
Simply Explained Inference Steps 

### ISSUE THREAD

**Issue by ndisci** [nii-yamagishilab/PartialSpoof#7]
Hello,

Thanks for the work. I tried to follow your "Using Pretrained Models" part of readme file in 03multireso folder. 

I am confused about your repository. I downloaded your models and datasets. But I could not understand the next steps. Could you explain it simply ? 

Where is your requirements.txt to create an environment ? 

Which version we should install for s3prl to work properly ?

How can we run main.py from CLI ? For instance what does --module-model , --model-forward-with-file-name or --module-config args mean? 

Thanks for your time.

**Comment by zlin0** [comment]
Hi, thanks for your interest in PartialSpoof.

For environment-related questions, please carefully read and follow the instructions at [03multireso#prepare](https://github.com/nii-yamagishilab/PartialSpoof/tree/main/03multireso#prepare) to build your environment. I didn't provide a requirements.txt because the environment for this repository fully follows the environment setup of [s3prl#installation](https://github.com/s3prl/s3prl/tree/7ab62aaf2606d83da6c71ee74e7d16e0979edbc3#installation), as mentioned in the Readme.md.

For inference, you can find the command line in stage 2 of 00_run.sh, which can also be used in the CLI.

For the meaning of --module-model, --model-forward-with-file-name, and --module-config, detailed explanations can be found in [arg_parse.py](https://github.com/nii-yamagishilab/PartialSpoof/blob/main/project-NN-Pytorch-scripts.202102/core_scripts/config_parse/arg_parse.py). These parameters follow the style of [project-NN-Pytorch-scripts](https://github.com/nii-yamagishilab/project-NN-Pytorch-scripts/tree/b91de2259bbcf095cadfe83a9a2e2e7129eec523). You can find a more detailed tutorial in this repository.

**Comment by ndisci** [comment]
Thanks for the reply. I have already taken a look arg_parse.py but could not find any detailed explanation like help strings.

**Comment by ndisci** [comment]
Finally I could run the main.py and now "data_gen" in nn_manager_zl.py is "None". Because of that I got "No output saved" warning. End of the process I just got output_score pkls.

**Comment by ndisci** [comment]
@zlin0 Could you help me about the issue ? :)

**Comment by zlin0** [comment]
> Thanks for the reply. I have already taken a look arg_parse.py but could not find any detailed explanation like help strings.

could you check again and be careful? There are helping infos.

**Comment by zlin0** [comment]
> Finally I could run the main.py and now "data_gen" in nn_manager_zl.py is "None". Because of that I got "No output saved" warning. End of the process I just got output_score pkls.


Could you provide more info on that? How did you run experiments? Did you check whether you loaded the model and data properly? Have you checked output_score pkls?

**Comment by ndisci** [comment]
Hi @zlin0 ,

I did run experiment by moving some folders in the project and changing the project structure.
I think you should add "--train-model" argument in run.sh for stage 2 in the CLI commands. 
When I followed s3prl environment setup , could not run the project. Because it needs extra packs. 
Now I am trying to run your metric calculation sh. To do this I will try to find a proper version for pyannote.metric.

Thanks for your time.

**Comment by ndisci** [comment]
Hi again, 

Sorry to bother you but I could not run RangeERR or SegmentERR script because of not existing some files (label2num_2cls_0sil for example.) Also how can we get rttm file ?

**Comment by zlin0** [comment]
> Sorry to bother you but I could not run RangeERR or SegmentERR script because of not existing some files (label2num_2cls_0sil for example.) Also how can we get rttm file ?

sorry, that's my fault that I only mentioned this in another issue but forgot to upload it to the repo. please go to this [answer](https://github.com/nii-yamagishilab/PartialSpoof/issues/1#issuecomment-1846922925) to download the [PS_data.tar.gz ](https://github.com/nii-yamagishilab/PartialSpoof/files/13612852/PS_data.tar.gz).

**Comment by zlin0** [comment]
> I did run experiment by moving some folders in the project and changing the project structure.

So please be careful when removing folders, as this might introduce some issues. This repo is based on some open-source projects like project-NN-Pytorch-scripts and s3prl.

> I think you should add "--train-model" argument in run.sh for stage 2 in the CLI commands.

Again, please read carefully. We already have `--module-model` for the same purpose, which you asked at the beginning. As I answered before, detailed explanations for `--module-model`, `--model-forward-with-file-name`, and `--module-config` can be found in [arg_parse.py](https://github.com/nii-yamagishilab/PartialSpoof/blob/main/project-NN-Pytorch-scripts.202102/core_scripts/config_parse/arg_parse.py). If you read carefully, you will find it helpful. Do you need me to copy it here?
```
    mes = 'module of model definition (default model, model.py will be loaded)'
    parser.add_argument('--module-model', type=str, default="model", help=mes)
```


> When I followed s3prl environment setup , could not run the project. Because it needs extra packs.

Please make sure you download the specific version and the needed extra packages if there are any. Sorry, but I have no time to prepare a perfect introduction for preparing the environment right now, but I am sure everything will become easier after you build the environment for [s3rpl#installation](https://github.com/s3prl/s3prl/tree/7ab62aaf2606d83da6c71ee74e7d16e0979edbc3#installation). 


> Now I am trying to run your metric calculation sh. To do this I will try to find a proper version for pyannote.metric.

Only RangeEER requires pyannote.metric, which provides `DetectionCostFunction().` pyannote.metric has not been updated in the last two years. So, please download the latest one.


And sorry, I am occupied with other projects, so I am not able to make my repo perfect for now, and also answer you questions immediately. However, some friends have successfully re-implemented this repo, so I believe all issues related to enviorments can be resolved even if there are any.

**Comment by ndisci** [comment]
Thank you so much ! :)
